### PR TITLE
Feat/fast eigen

### PR DESCRIPTION
**Pull request by giovannic**
Implement fast leading eigenvalue calculation. Using power iteration

### CODE REVIEW

File: src/eigenvalues.js [hunk 1]
```diff
@@ -1,115 +1,37 @@
...
-            for (let x = 0; x < k; x++) {
-                m[idx(x, y)] = sum(ry, (v, j) => v * q[idxq(x, j + y)]);
-            }
-        }
-        
-        // (option) post process M = M + mu * I
-        for (let i = 0; i < k; i++) m[idx(i, i)] += mu;        
+function powerIteration(m, n = 100, error = 1e-6) {
+  const max_m = max(m)
+  let x = Array(m[0].length).fill(0).map(() => { return random(1, max_m) });
```
Comment (richfitz) [inline src/eigenvalues.js:20]: is it worth adding the reason for this here (I had to look it up - looks like we want to decrease the chance we pick an orthogonal vector, which must be vanishingly unlikely with a matrix of this size)

File: test/test_eigen.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('leadingEigenvalue', function() {
+  it('works for a non-symmetric matrix', function() {
+    expect(
+      leadingEigenvalue([[1, 2], [3, 4]])
+    ).to.be.closeTo(5.3722813, 1e-6);
+  });
+
+  it('is fast', function() {
+    this.timeout(1000);
```
Comment (richfitz) [inline test/test_eigen.js:14]: I had a little fight but could not get the old version to easily run here as a comparison - how much faster is this?

**Review by richfitz — APPROVED** [review]
LGTM, just a couple of queries/comments

File: test/test_eigen.js (hunk 2 above)
Comment (giovannic) [reply test/test_eigen.js:14]: Ah, so the build before did the Brazil test in [6.4s](https://github.com/mrc-ide/squire_js/runs/1459029323?check_suite_focus=true#step:4:80)
And this build does it in [.4s](https://github.com/mrc-ide/squire_js/runs/1464364460?check_suite_focus=true#step:4:81)

So probably ~16x?

**Pull request merged**